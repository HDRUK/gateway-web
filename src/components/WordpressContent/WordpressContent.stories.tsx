import type { Meta, StoryObj } from "@storybook/react";
import WordpressContent from "@/components/WordpressContent";

const meta: Meta<typeof WordpressContent> = {
    component: WordpressContent,
    tags: ["autodocs"],
};

export default meta;

type Story = StoryObj<typeof WordpressContent>;

export const Default: Story = {
    args: {
        content:
            "\n<p>This month we are pleased to share improvements to the usability of the search results page. Plus, we release our data use register widget.</p>\n",
    },
};
