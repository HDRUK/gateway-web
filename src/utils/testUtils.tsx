import React, { ReactNode } from "react";
import {
    render,
    renderHook,
    RenderHookOptions,
    RenderOptions,
    RenderResult,
    RenderHookResult,
} from "@testing-library/react";
import { SWRConfig } from "swr";
import { CacheProvider, ThemeProvider } from "@emotion/react";
import ActionBarProvider from "@/providers/ActionBarProvider";
import DialogProvider from "@/providers/DialogProvider";
import theme from "@/config/theme";
import createEmotionCache from "@/config/createEmotionCache";

const clientSideEmotionCache = createEmotionCache();

const Wrapper = ({ children }: { children: ReactNode }) => {
    return (
        <SWRConfig
            value={{
                provider: () => new Map(),
            }}>
            <CacheProvider value={clientSideEmotionCache}>
                <ThemeProvider theme={theme}>
                    <ActionBarProvider>
                        <DialogProvider>{children}</DialogProvider>
                    </ActionBarProvider>
                </ThemeProvider>
            </CacheProvider>
        </SWRConfig>
    );
};

interface OptionProps
    extends RenderOptions<
        typeof import("@testing-library/dom/types/queries"),
        HTMLElement,
        HTMLElement
    > {
    wrapperProps: Record<string, unknown>;
}

const customRender = (
    ui: React.ReactElement<
        unknown,
        string | React.JSXElementConstructor<unknown>
    >,
    options?: OptionProps
): RenderResult => {
    const { wrapperProps, ...rest } = options || {};

    return render(ui, {
        wrapper: props => <Wrapper {...props} {...wrapperProps} />,
        ...rest,
    });
};

interface OptionHookProps
    extends RenderHookOptions<
        unknown,
        typeof import("@testing-library/dom/types/queries"),
        HTMLElement,
        HTMLElement
    > {
    wrapperProps: Record<string, unknown>;
}

const customRenderHook = (
    ui: (initialProps: unknown) => unknown,
    options?: OptionHookProps
): RenderHookResult<unknown, unknown> => {
    const { wrapperProps, ...rest } = options || {};
    return renderHook(ui, {
        wrapper: props => <Wrapper {...props} {...wrapperProps} />,
        ...rest,
    });
};

export * from "@testing-library/react";

export { customRenderHook as renderHook };
export { customRender as render };
