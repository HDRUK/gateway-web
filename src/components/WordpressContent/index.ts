import WordpressContent from "./WordpressContent";

export default WordpressContent;
