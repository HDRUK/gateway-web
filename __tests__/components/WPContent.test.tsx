import React from "react";
import WordpressContent from "@/components/WordpressContent";
import { render } from "../testUtils";

describe("WordpressContent", () => {
    it("should render component", async () => {
        const html =
            "\n<p>This month we are pleased to share improvements to the usability of the search results page. Plus, we release our data use register widget.</p>\n";

        const wrapper = render(<WordpressContent content={html} />);

        expect(wrapper.container).toMatchSnapshot();
    });
});
