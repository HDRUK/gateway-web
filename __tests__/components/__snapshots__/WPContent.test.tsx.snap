// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WPContent should render component 1`] = `
<div>
  <div
    class="mui-style-y30c9z-WPContent"
  >
    

    <p>
      This month we are pleased to share improvements to the usability of the search results page. Plus, we release our data use register widget.
    </p>
    

  </div>
</div>
`;
